react/features/base/conference/functions.ts

```diff
@@ -272,21 +272,54 @@ export function generateVisitorConfig(stateful: IStateful, params: Array<string>
         return;
     }
 
-    const oldDomain = config.hosts.domain;
+    if (!vnode && config.oldConfig && username) {
```

Comment:
Is `username` required when redirecting back? The case of `vnode == undefined && username == undefined` will just fall back to the redirect code below, which is probably not what you want.

I suggest something like 
```
if (!vnode) { 
  // check config.oldConfig and username and log/throw/return an error
  // return the restored config
} else {
  // return the vnode config
} 
```